geode-core/src/main/java/org/apache/geode/internal/cache/AbstractUpdateOperation.java

```diff
@@ -169,7 +169,7 @@ public static boolean doPutOrCreate(LocalRegion rgn, EntryEventImpl ev, long las
       // from this message.
       if (doUpdate) {
         if (!ev.isLocalInvalid()) {
-          final long startPut = CachePerfStats.getStatTime();
+          final long startPut = rgn.getCachePerfStats().getTime();
```

Comment:
What makes this preferable to `rgn.getStatisticsClock().getTime()`? (I realize that would violate your admonition in the `AbstractRegion.getStatisticsClock()` javadoc, but I don't yet understand the reason for that admonition.)

Reply:
There are two reasons. First, I wanted to limit `AbstractRegion.getStatisticsClock()` to being invoked only by subclasses of `AbstractRegion` but unfortunately `protected` won't protect it since everything under the sun lives in the package `cache`.

Also, many classes already need to invoke `region.getCachePerfStats()` so it's a well known accessor unlike the new `getStatisticsClock()`. The following pattern is really common throughout our code (especially `Operation` classes):
```
long start = rgn.getCachePerfStats().getTime();
try {
  // perform operation
} finally {
  rgn.getCachePerfStats().endOperation(start);
}
```
Some of the operations also follow the following variation, which isn't as clear but if you know of the above then it becomes natural to use `rgn.getCachePerfStats().getTime()`:
```
long start = rgn.getCachePerfStats().getTime();
performOperation(start);
```
I realize it would be cleaner to inject `CachePerfStats` into each of these classes that need to start/end stats that are defined in `CachePerfStats and that's exactly what I'd like to see happen (just not for this PR which is already quite long).

Reply:
Note: I also kept the list of calls to `cache.getStatisticsClock()` very slim on purpose. I tried to invoke that just when constructing and passing in `StatisticsClock` to various factory type classes, entities that the cache owns, and entities that the cache is essentially a factory for (such as regions).

Reply:
Got it. Thanks for the explanations.